taqueria-plugin-tezos-client/client.ts

```diff
@@ -0,0 +1,256 @@
+}
+
+const typecheck = <T>(parsedArgs: Opts, sandbox: Sandbox): LikeAPromise<PluginResponse, Failure<T>> => {
+    const sourceFiles = (parsedArgs.sourceFiles as string).split(',')
+    let p;
+    if (parsedArgs.sourceFiles) {
+        if (sourceFiles.length == 1)
+            p = typecheckContract (parsedArgs, sandbox) (parsedArgs.sourceFiles as string).then(data => [data])
+        else
+            p = typecheckMultiple (parsedArgs, sandbox) (sourceFiles)
+    } else {
+        p = typecheckAll (parsedArgs, sandbox)
+            .then(results => {
+                if (results.length === 0) sendErr("No contracts found to compile.")
+                return results
+            })
+    }
+    return p.then(sendAsyncJsonRes)
+}
+
+const typecheckTask = async <T>(parsedArgs: Opts) : LikeAPromise<void, Failure<T>> => {
+    const sandbox = getSandbox(parsedArgs)
+    if (sandbox) {
+        if (doesUseFlextesa(sandbox)) {
+            return await isSandboxRunning(sandbox.name.value)
+            ? typecheck(parsedArgs, sandbox).then(sendJsonRes)
+            : sendAsyncErr(`The ${sandbox.name} sandbox is not running.`)
+        }
+        return sendAsyncErr(`Cannot start ${sandbox.label} as its configured to use the ${sandbox.plugin} plugin.`)
+    }
+    return sendAsyncErr(`There is no sandbox configuration with the name ${parsedArgs.sandboxName}.`)
+}
+
+//////////// Simulate task ////////////
+
+const getStorageFromConfig = (opts: Opts, sourceFile: string) : string|undefined => {
+    const config = opts.config
+    const defaultEnv = config.environment.default
+    const storages = config.environment[defaultEnv].storage
+    const storageValue = storages[sourceFile]
+    return storageValue ? storageValue as string : undefined
+}
+
+const getSimulateCommand = (opts: Opts, sandbox: Sandbox, sourceFile: string, sourcePath: string) => {
+    let storage: string|undefined
+    if (opts.storage) {
+        if (typeof opts.storage === 'string') storage = opts.storage as string
+        else storage = `${opts.storage}`
+    } else {
+        storage = getStorageFromConfig(opts, sourceFile)
+        if (!storage) throw new Error('Error: Please specify a non-empty storage value in the CLI or in the config file.')
+    }
+
+    let input: string = opts.input && typeof opts.input === 'string' ? opts.input as string : `${opts.input}`
+
+    // If the string contains escaped double quotes, escape them further
+    storage = storage.replace(/\\"/g, '\\\\\\"')
+    input = input.replace(/\\"/g, '\\\\\\"')
+
+    // If the string contains leading and trailing double quotes, escape them, otherwise docker exec will complain
+    storage = storage.match(/^".*"$/) ? "\\" + storage.slice(0, -1) + "\\\"" : storage
+    input = input.match(/^".*"$/) ? "\\" + input.slice(0, -1) + "\\\"" : input
+
+    // TODO: maybe validate storage and input value before passing it to tezos-client to prevent tezos-client menu being displayed?
```

Comment:
I think we can either trim everything down starting from "Usage" or do data validation as we go (@mweichert suggests the latter if I recall). Since it's pretty isolated, I suppose it could be a separate "enhancement issue"?

Reply:
Given the low chance of someone actually encountering this, a quick and dirty solution is fine here for now.

If we defer the validation work, let's make sure to log the in a separate ticket

Reply:
hmm on second thought, I feel like trimming seems to be a much easier solution and it covers all cases (have `tezos-client` do all the validation so that we are not re-inventing the wheel).

Reply:
I think trimming it is the best solution for now

Reply:
Thoughts? @mweichert

Reply:
I implemented trimming and it works like wonders! I think we should stick with this. It's also more maintainable because we only have to change a single line in case `tezos-client` change their menu to removing "Usage:" (it's the pattern I look for when I trim).